micro-rdk/src/common/credentials_storage.rs

```diff
@@ -300,6 +305,15 @@ impl WifiCredentialStorage for RAMStorage {
     }
 }
 
+impl StorageDiagnostic for RAMStorage {
+    type Error = Infallible;
+    fn log_space_diagnostic(&self) -> Result<(), Self::Error> {
+        let used_space = size_of_val(self);
+        log::info!("CREDENTIAL STORAGE STATS: {:?} bytes used", used_space);
```

Comment:
does this just return the size of the `Rc<Mutex<>>`? or does it actually change depending on our usage in RAMStorage?

Reply:
I checked and you're right, this doesn't give the right size